spec/requests/api/v1/spots/create_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+describe 'POST api/v1/spots', type: :request do
+  shared_examples 'invalid params' do
+    it 'does not create the spot' do
+      expect do
+        post api_v1_spots_path, params: params, headers: auth_headers, as: :json
+      end.not_to change(Spot, :count)
+    end
+
+    it 'does not return a successful response' do
+      post api_v1_spots_path, params: params, headers: auth_headers, as: :json
+      expect(response).to have_http_status(:bad_request)
+    end
+  end
+
+  let!(:user) { create(:user) }
+  let(:spot)  { Spot.last }
+
+  context 'with correct params' do
+    let!(:topic) { create(:topic) }
+    let(:params) do
+      {
+        spot: attributes_for(:spot, topic_id: topic.id)
+      }
+    end
+
+    it 'creates the spot' do
+      expect do
+        post api_v1_spots_path, params: params, headers: auth_headers, as: :json
+      end.to change(Spot, :count).by(1)
+    end
+
+    it 'returns a succesful response' do
+      post api_v1_spots_path, params: params, headers: auth_headers, as: :json
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'returns the spot' do
+      post api_v1_spots_path, params: params, headers: auth_headers, as: :json
+      expect(json[:spot]).to include_json(
+        id: spot.id,
+        latitude: format('%.6g', format('%.6f', spot.latitude)),
+        longitude: format('%.6g', format('%.6f', spot.longitude)),
+        radius: format('%.1f', spot.radius),
+        topic_id: spot.topic_id
+      )
+    end
+  end
+  context 'with incorrect params' do
+    context 'when title is blank' do
+      let!(:topic) { create(:topic) }
```

Comment:
Move this sentence to line 16 and remove it from the examples below. Isn't necessary to create a topic for each example, is the same case than the user remember?